plugin/trino-vertica/src/main/java/io/trino/plugin/vertica/VerticaClient.java

```diff
@@ -0,0 +1,248 @@
+            case Types.CHAR:
+                return Optional.of(charColumnMapping(createCharType(typeHandle.getRequiredColumnSize()), false));
+            case Types.VARCHAR:
+            case Types.LONGVARCHAR:
+                return Optional.of(varcharColumnMapping(createVarcharType(typeHandle.getRequiredColumnSize()), false));
+            case Types.DATE:
+                return Optional.of(dateColumnMapping());
+        }
+        if (getUnsupportedTypeHandling(session) == CONVERT_TO_VARCHAR) {
+            return mapToUnboundedVarchar(typeHandle);
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    public WriteMapping toWriteMapping(ConnectorSession session, Type type)
+    {
+        if (type == BOOLEAN) {
+            return WriteMapping.booleanMapping("boolean", booleanWriteFunction());
+        }
+        if (type == SMALLINT) {
+            return WriteMapping.longMapping("smallint", smallintWriteFunction());
+        }
+        if (type == TINYINT) {
+            return WriteMapping.longMapping("tinyint", tinyintWriteFunction());
+        }
+        if (type == INTEGER) {
+            return WriteMapping.longMapping("integer", integerWriteFunction());
+        }
+        if (type == BIGINT) {
+            return WriteMapping.longMapping("bigint", bigintWriteFunction());
+        }
+        if (type == REAL) {
+            return WriteMapping.longMapping("real", realWriteFunction());
+        }
+        if (type == DOUBLE) {
+            return WriteMapping.doubleMapping("double precision", doubleWriteFunction());
+        }
+        if (type instanceof DecimalType) {
+            DecimalType decimalType = (DecimalType) type;
+            String dataType = format("decimal(%s, %s)", decimalType.getPrecision(), decimalType.getScale());
+            if (decimalType.isShort()) {
+                return WriteMapping.longMapping(dataType, shortDecimalWriteFunction(decimalType));
+            }
+            return WriteMapping.sliceMapping(dataType, longDecimalWriteFunction(decimalType));
+        }
+        if (type instanceof CharType) {
+            int length = ((CharType) type).getLength();
+            checkArgument(length <= 65000, "Char length is greater than 65,000");
+            return WriteMapping.sliceMapping("char(" + length + ")", charWriteFunction());
+        }
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            String dataType;
+            if (varcharType.isUnbounded()) {
+                dataType = "long varchar";
+            }
+            else {
+                checkArgument(varcharType.getBoundedLength() <= 32_000_000, "Varchar length is greater than 32,000,000");
+                dataType = "long varchar(" + varcharType.getBoundedLength() + ")";
+            }
+            return WriteMapping.sliceMapping(dataType, varcharWriteFunction());
```

Comment:
Wouldn't this error out when length > 65k?

Reply:
Also it looks like we aren't handling cases where the value has multi-bytes characters? We can tackle this later but add a TODO comment in that case.

i.e. with a `VARCHAR(1)` with value `U+1F600` (3 bytes) will not fit in a `VARCHAR(1)` in Vertica.
Trino counts codepoints/characters while Vertica counts bytes/octets.